piptools/scripts/compile.py

```diff
@@ -143,6 +143,8 @@ def cli(verbose, dry_run, pre, rebuild, find_links, index_url, extra_index_url,
         repository = LocalRequirementsRepository(existing_pins, repository)
 
     log.debug('Using indexes:')
+    # remove duplicate index urls before processing
+    repository.finder.index_urls = sorted(set(repository.finder.index_urls))
```

Comment:
That's fine by me. I have pushed an update with the collections implementation as per your link.